_rules/table-headers-attribute-refer-to-data-cells-a25f45.md

```diff
@@ -0,0 +1,318 @@
+	</tbody>
+</table>
+```
+
+#### Passed Example 2
+
+The `headers` attribute on the cell refers to a `th` element within the same `table`. Multiple headers are referenced for a cell with `colspan` of `2`.
+
+```html
+<table>
+	<thead>
+		<tr>
+			<th id="header1">Projects</th>
+			<th id="header2">Exams</th>
+		</tr>
+	</thead>
+	<tbody>
+		<tr>
+			<td colspan="2" headers="header1 header2">15%</td>
+		</tr>
+	</tbody>
+</table>
+```
+
+#### Passed Example 3
+
+The `headers` attribute on the second cell refers to a `td` element with a role of `columnheader` within the same `table`.
+
+```html
+<table>
+	<thead>
+		<tr>
+			<td role="columnheader" id="header1">Projects</td>
+		</tr>
+	</thead>
+	<tbody>
+		<tr>
+			<td headers="header1">15%</td>
+		</tr>
+	</tbody>
+</table>
+```
+
+#### Passed Example 4
+
+This `table` has multiple `columnheader(s)`. The `headers` attribute on the cells lists the `id(s)` of `th` elements within the same `table`.
+
+```html
+<table>
+	<tr>
+		<th colspan="2" id="header1">Projects</th>
+		<th colspan="2" id="header2">Exams</th>
+	</tr>
+	<tr>
+		<th id="e1" headers="header1">1</th>
+		<th id="e2" headers="header1">2</th>
+		<th id="p1" headers="header2">1</th>
+		<th id="p2" headers="header2">2</th>
+	</tr>
+	<tr>
+		<td colspan="2" headers="header1 e1 e2">15%</td>
+		<td headers="header2 p1">15%</td>
+		<td headers="header2 p2">45%</td>
+	</tr>
+</table>
+```
+
+#### Passed Example 5
+
+The `headers` attribute on the cell refers to a `th` element with a role of `rowheader` within the same `table`.
+
+```html
+<table>
+	<tbody>
+		<tr>
+			<th role="rowheader" id="headerAge">Age</th>
+			<td headers="headerAge">65</td>
+		</tr>
+	</tbody>
+</table>
+```
+
+#### Passed Example 6
+
+The `headers` attribute on the last two `th` elements refers to another `th` element within the same `table`. Here the column header has a span of two columns.
+
+```html
+<table>
+	<tr>
+		<th id="name" colspan="2">Name</th>
+	</tr>
+	<tr>
+		<th headers="name">Firsname</th>
+		<th headers="name">Lastname</th>
+	</tr>
+</table>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The `headers` attribute on the cell refers to an element outside the same `table`.
+
+```html
+<span id="elmOutsideTable">Project Costs</span>
+
+<table>
+	<tr>
+		<th id="headerOfColumn">Projects</th>
+	</tr>
+	<tr>
+		<td headers="elmOutsideTable">15%</td>
+	</tr>
+</table>
+```
+
+#### Failed Example 2
+
+The `td` element has a `headers` attribute referring to an ID that does not exist within the same `table` (or within the [document](https://dom.spec.whatwg.org/#concept-document)). Even if some of the ID referred to exist, a single non-existing one is enough to fail the rule.
+
+```html
+<table>
+	<tr>
+		<th colspan="2" id="header1">Projects</th>
+	</tr>
+	<tr>
+		<th id="e1" headers="header1">1</th>
+		<th id="e2" headers="header1">2</th>
+	</tr>
+	<tr>
+		<td colspan="2" headers="header1 e1 NOT-EXIST">15%</td>
+	</tr>
+</table>
+```
+
+#### Failed Example 3
```

Comment:
This is quite redundant with Failed Example 2, but simplified. Could be removed.